src/messages.c

```diff
@@ -0,0 +1,274 @@
+#include "bouncer.h"
+
+/* Inspect Parse packet to see if it defines a named prepared statement */
+bool inspect_parse_packet(PgSocket *client, PktHdr *pkt, uint8_t *ps_action)
+{
+	const char *statement;
+	const char *query;
+	uint16_t num_parameters;
+
+	if (!mbuf_get_string(&pkt->data, &statement))
+		return false;
+
+	if (!mbuf_get_string(&pkt->data, &query))
+		return false;
+
+	/* number of parameter data types */
+	if (!mbuf_get_uint16be(&pkt->data, &num_parameters))
+		return false;
+
+	if (strlen(statement) > 0) {
+		slog_noise(client, "inspect_parse_packet: type=%c, len=%d, statement=%s", pkt->type, pkt->len, statement);
+		*ps_action = PS_HANDLE;
+	} else {
+		slog_noise(client, "inspect_parse_packet: type=%c, len=%d, statement=<empty>", pkt->type, pkt->len);
+		*ps_action = PS_IGNORE;
```

Comment:
For all `inspect_xxx_packet` functions: Let's move the check for empty statement right after we get the `statement` from the `pkt`. Then we can do an early return there. No need to do any more packet parsing when that happens. 

The underlying reason for this comment is that I would really like to have this feature on by default, if at all possible. So I'd like to make sure there's no significant perf impact when named prepared statements are not used, even when they are enabled. So it would be good not to spend extra cycles once we know that it's not a named prepared statement.

Reply:
Done